Use `torch.det` to calculate volumes

tsihyoung opened the pull request:
## Summary

`torch.cross` without `dim` is deprecated.

## Todos

I'm not sure whether we need a `torch.abs` here.

janosh reviewed (approved):
`torch.det` was the last tensor op in chgnet lacking MPS support which is [why we refactored to](https://github.com/CederGroupHub/chgnet/commit/4cee9e6aaa8fbe27a6ab9fd583d797e234cd3280) `torch.dot(lattice[0], torch.cross(lattice[1], lattice[2]))` to support MPS earlier. based on tests passing on `macos-14`, looks like `torch.det` is now implemented in MPS so happy to merge.

tsihyoung commented:
> `torch.det` was the last tensor op in chgnet lacking MPS support which is [why we refactored to](https://github.com/CederGroupHub/chgnet/commit/4cee9e6aaa8fbe27a6ab9fd583d797e234cd3280) `torch.dot(lattice[0], torch.cross(lattice[1], lattice[2]))` to support MPS earlier. based on tests passing on `macos-14`, looks like `torch.det` is now implemented in MPS so happy to merge.

I tested `torch.det` on my Apple Silicon machine with `pytorch == 2.2.1`. Unfortunately, it still complains that`NotImplementedError: The operator 'aten::_linalg_det.result' is not currently implemented for the MPS device.`

Currently, maybe we should change the code to `torch.dot(lattice[0], torch.cross(lattice[1], lattice[2], dim=-1))`.

janosh commented:
interesting, that means the tests on `macos-14` aren't actually using MPS. wonder why...
